modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java

```diff
@@ -626,6 +626,9 @@ private void updateComponentSize() {
         newScaleFactorY = GraphicsEnvironment.getLocalGraphicsEnvironment().
                           getDefaultScreenDevice().getDefaultConfiguration().
                           getDefaultTransform().getScaleY();
+        if (oldWidth == 0 && oldHeight == 0 && pWidth == 0 && pHeight == 0) {
+            return;
+        }
```

Comment:
this change fixes the issue with the provided test code.

a few comments:
1. if I wrap textArea with a BorderPane, I never get width=0 (master branch)
2. would it make sense to add a check to line 611: if((pWidth == 0) && (pHeight == 0))?
3. lines 617, 618 subtract border insets, would that risk making pWidth/pHeight negative (considering code on lines 613, 614)?

Reply:
I guess l618, 619 in PR takes care of pt 2,3 concern..